lib/ansible/modules/network/nxos/nxos_hsrp.py

```diff
@@ -239,6 +241,19 @@ def get_hsrp_group(group, interface, module):
     return hsrp
 
 
+def get_hsrp_group_unknown_enum(module, command, hsrp_table):
+    '''Some older NXOS images fail to set the attr values when using structured output and
+    instead set the values to <unknown enum>. This fallback method is a workaround that
+    uses an unstructured (text) request to query the device a second time.
+    'sh_preempt' is currently the only attr affected. Add checks for other attrs as needed.
+    '''
+    if 'unknown enum:' in hsrp_table['sh_preempt']:
+        cmd = {'output': 'text', 'command': command.split('|')[0]}
```

Comment:
We actually have the code already implemented in module_utils https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/nxos/nxos.py#L164.
When `check_rc` is set to `retry_json`, run_commands will send the command the request with non structured format if json fails. 
You will just need to append this error `unknown enum` in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/nxos/nxos.py#L170